src/drug_discovery/structures/protein.py

```diff
@@ -501,6 +534,34 @@ def remove_water(self) -> None:
         """
         self.structure = self.structure[~filter_solvent(self.structure)]
 
+    def find_missing_residues(self) -> dict:
+        """find missing residues in the protein structure"""
+
+        from Bio.PDB import PDBParser
+
+        parser = PDBParser(QUIET=True)
+        structure = parser.get_structure("protein", self.file_path)
+
+        missing = {}
+
+        for model in structure:
+            for chain in model:
+                chain_id = chain.id
+                last_resseq = None
+                gaps = []
+
+                residues = [res for res in chain.get_residues() if res.id[0] == " "]
```

Comment:
Depending on existing guarantees on input, maybe worth sorting:
```
residues = sorted(
    [res for res in chain.get_residues() if res.id[0] == " "],
    key=lambda r: r.id[1]
)
```

But I might lack context, so validate.

Reply:
that's great! fixed